core/extension/gdextension_interface.cpp

```diff
@@ -1375,6 +1377,19 @@ static void gdextension_editor_remove_plugin(GDExtensionConstStringNamePtr p_cla
 #endif
 }
 
+static void gdextension_editor_help_load_xml_from_utf8_chars_and_len(const char *p_data, GDExtensionInt p_size) {
+#ifdef TOOLS_ENABLED
+	EditorHelp::load_xml_buffer((const uint8_t *)p_data, p_size);
```

Comment:
There is going to be one tricky thing to solve here: code in `core/` is not supposed to directly call code in `editor/`.

We've solved this in a couple different ways in the past. For example, we couldn't directly call `EditorNode::add_plugin()` to register editor plugins. Instead, `EditorNode` sets some function pointers on `GDExtensionEditorPlugin`, so then `gdextension_interface.cpp` can call them.

Another example was with extension reloading: after it's finished, we need to refresh the editor inspector. So, we added the `extensions_reloaded` signal on `GDExtensionManager`, and `EditorNode` connects to that signal.

I'm not totally sure what the right thing to do in this case is, though. I don't think we'd want to do a signal, because we don't want scripts to be able to connect to it. So, maybe a solution closer to my first example would work best? Maybe we could have a static class named like `GDExtensionEditorHelp` which has some function pointers that the real `EditorHelp` could set when it's initialized?

Reply:
> Out of curiosity, why's this exactly? To avoid some sort of logical circular dependency, for cleanliness or something else?

I think for avoiding circular dependencies? We may even have this rule written down in the documentation somewhere but I couldn't find it just now. But the principle is that "higher level" things can depend on "lower level" things, but not the other way around. It goes something like Editor -> Scene -> Servers -> Core, so parts can depend on things to their right but not their left.

> Edit: oh I'm noticing that it's not building properly in the CI. I suppose it's related to that?

Possibly? But you may just be missing an `#include`